src/main.c

```diff
@@ -799,6 +830,7 @@ static gchar* r_status_formatter_shell(void)
 		formatter_shell_append_n(text, "RAUC_SLOT_BOOTNAME", slotcnt, slot->bootname);
 		formatter_shell_append_n(text, "RAUC_SLOT_PARENT", slotcnt, slot->parent ? slot->parent->name : NULL);
 		formatter_shell_append_n(text, "RAUC_SLOT_MOUNTPOINT", slotcnt, slot->mount_point);
+		formatter_shell_append_n(text, "RAUC_SLOT_BOOT_STATUS", slotcnt, slot->bootname ? good ? "good" : "bad" : NULL);
```

Comment:
Avoid the nested ternary operator by making this conditional on the the existence of slot->bootname.